gordon_janitor_gcp/msg_generators.py

```diff
@@ -0,0 +1,70 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2018 Spotify AB
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Module to collect classes that transform different inputs into messages that
+can be passed from one plugin to another through a channel.
+
+Example A record message generator usage:
+
+.. code-block:: pycon
+
+    >>> import asyncio
+    >>> loop = asyncio.get_event_loop()
+    >>> instance1_data = {'hostname': 'a-host', 'internal_ip': '192.168.11.5'}
+    >>> instance2_data = {'hostname': 'b-host', 'internal_ip': '192.168.11.6'}
+    >>> generator = ARecordMsgGenerator('example.com')
+    >>> batch = loop.run_until_complete(
+            generator.generate_batch([instance1_data, instance2_data]))
+    >>> print(batch)
+    {'zone': 'example.com', 'rrsets':
+        [{'name': 'a-host', 'type': 'A', 'rrdatas': ['192.168.11.5']},
+         {'name': 'b-host', 'type': 'A', 'rrdatas': ['192.168.11.6']}]}
+"""
+
+
+# TODO: uncomment the interface decorator one interface PR is merged in.
+# @zope.interface.implementer(interfaces.IMessageGenerator)
+class ARecordMsgGenerator:
+    """Generates messages used to verify A records.
+
+    Args:
+        zone (str): DNS zone that the instance's record should belong to.
+    """
+    def __init__(self, zone):
+        self.zone = zone
+
+    async def generate_msg(self, instance):
+        """Transform instance data into a dictionary.
+
+        Args:
+            instance (dict): GCP instance data.
+        Returns:
+            dict: instance data formatted for submitting to a channel.
+        """
+        return {
+            'name': instance.get('hostname'),
+            'type': 'A',
+            'rrdatas': [instance.get('internal_ip')]
```

Comment:
Wouldn't the common use case be for external IP instead of internal? I don't know if internal is more of a spotify thing (it may not be), but I feel like there'd definitely be a common need for external.